src/accessors/IIterator.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Created by oliverjaegle on 22.06.17.
+ */
+
+export interface IIterator<T> {
+    next(): Array<T>;
+
+    hasMoreElements(): boolean;
```

Comment:
@graywolf336 This way, the iterator implementation needs a state. But it allows `next` to return a more simple typed result